extensions/yaml_support/labgraph_monitor/generate_lg_monitor/generate_lg_monitor.py

```diff
@@ -0,0 +1,215 @@
+            for edge in stream.topic_paths:
+                if edge != upstream_edge:
+                    in_out_edge_map[edge] = upstream_edge
+
+    return in_out_edge_map
+
+
+def connect_to_upstream(
+    nodes: List[LabgraphMonitorNode],
+    streams: Stream
+) -> List[LabgraphMonitorNode]:
+    """
+    A function that connect a node to its upstream node
+
+    @params:
+        nodes: The list of nodes of the graph
+        streams: sequence of messages that are accessible in real-time
+
+    @return: The new list of nodes after update
+    """
+    out_edge_node_map = out_edge_node_mapper(nodes)
+    in_out_edge_map = in_out_edge_mapper(streams)
+
+    for node in nodes:
+        if node.in_edge:
+            node.upstream_node = out_edge_node_map[
+                in_out_edge_map[node.in_edge]
+            ]
+
+    return nodes
+
+
+def serialize_graph(
+    name: str,
+    nodes: List[LabgraphMonitorNode]
+) -> SerializedGraph:
+    """
+    A function that returns a serialized version of the graph topology.
+
+    @params:
+        name: The name of the graph
+        nodes: The list of nodes of the graph
+
+    @return: A serialized version of the graph topology
+    """
+    serialized_graph: SerializedGraph = {
+        "name": name,
+        "nodes": {}
+    }
+
+    for node in nodes:
+        if node.grouping not in serialized_graph["nodes"]:
+            serialized_graph["nodes"][node.grouping] = {
+                "upstreams": {}
+            }
+
+        if node.upstream_node:
+
+            if node.upstream_node.grouping not in \
+             serialized_graph["nodes"][node.grouping]["upstreams"]:
+                (serialized_graph["nodes"]
+                    [node.grouping]
+                    ["upstreams"]
+                    [node.upstream_node.grouping]) = [
+                        node.upstream_message.serialize()
+                    ]
+            else:
+                (serialized_graph["nodes"]
+                    [node.grouping]
+                    ["upstreams"]
+                    [node.upstream_node.grouping]).append(
+                        node.upstream_message.serialize()
+                    )
+
+    return serialized_graph
+
+
+def generate_labgraph_monitor(graph: lg.Graph) -> None:
+    """
+    A function that serialize the graph topology
+    and send it using to LabGraphMonitor Front-End
+    using Labgraph Websocket API
+
+    @params:
+        graph: An instance of the computational graph
+    """
+    # Local variables
+    nodes: List[LabgraphMonitorNode] = []
+
+    # Identify graph nodes
+    nodes = identify_graph_nodes(graph)
+
+    # Connect graph edges
+    nodes = connect_to_upstream(nodes, graph.__streams__.values())
+
+    # Serialize the graph topology
+    serialized_graph = serialize_graph(
+        type(graph).__name__,
+        nodes
+    )
+
+    # Send the serialized graph to Front-End
+    # using LabGraph Websockets API
+    run_server(serialized_graph)
```

Comment:
> Just to be sure @bennaaym, does this `SERIALIZER` publish:
> 
> * The current state of the graph (such as it is running the `Demo` graph `lg.run(Demo)`  as well as the `WSSenderNode` graph `lg.run(WSSenderNode)`)
> * Or does it just publish the data that it received from `serialize_graph()` without running `lg.run(Demo)`
> 
> I think there might be a small issue with the fact that it is not running the `Demo` graph while running the `WSSenderNode`, which makes it impossible to fetch real-time data right now

My question above is also related to this particular line, such as the `run_server()` function here is called with the already prepared version of the `seralized_graph` message,  **which does not update if the graph itself updates (i.e. the message data changes)**

```python
{
  'name': 'Demo',
  'nodes': {
    'NoiseGenerator': {
      'upstreams': {
        
      }
    },
    'RollingAverager': {
      'upstreams': {
        'NoiseGenerator': [
          {
            'name': 'RandomMessage',
            'fields': {
              'timestamp': {
                'type': 'float'
              },
              'data': {
                'type': 'ndarray'
              }
            }
          }
        ]
      }
    },
    'Amplifier': {
      'upstreams': {
        'NoiseGenerator': [
          {
            'name': 'RandomMessage',
            'fields': {
              'timestamp': {
                'type': 'float'
              },
              'data': {
                'type': 'ndarray'
              }
            }
          }
        ]
      }
    },
    'Attenuator': {
      'upstreams': {
        'NoiseGenerator': [
          {
            'name': 'RandomMessage',
            'fields': {
              'timestamp': {
                'type': 'float'
              },
              'data': {
                'type': 'ndarray'
              }
            }
          }
        ]
      }
    },
    'Sink': {
      'upstreams': {
        'RollingAverager': [
          {
            'name': 'RandomMessage',
            'fields': {
              'timestamp': {
                'type': 'float'
              },
              'data': {
                'type': 'ndarray'
              }
            }
          }
        ],
        'Amplifier': [
          {
            'name': 'RandomMessage',
            'fields': {
              'timestamp': {
                'type': 'float'
              },
              'data': {
                'type': 'ndarray'
              }
            }
          }
        ],
        'Attenuator': [
          {
            'name': 'RandomMessage',
            'fields': {
              'timestamp': {
                'type': 'float'
              },
              'data': {
                'type': 'ndarray'
              }
            }
          }
        ]
      }
    }
  }
}
```

Reply:
yeah, that's true, actually, the whole serialization process ( identify_graph_nodes, connect_to_upstream, serialize_graph) should be moved  from **generate_labgraph_monitor** to the SerializerNode. currently I'm only sending the result which was ok for only sending the topology without real-time values

Reply:
could you please let me know what you think of this approach? 

https://github.com/facebookresearch/labgraph/pull/64#issuecomment-1082286247

Reply:
@dtemir great work! I went through your current solution and it looks pretty promising however I can see a few challenges related to the current approach:
* The current approach is changing the topology of the main graph ( the graph to be serialized and streamed). for example, Serializer and WSAPIServerNode which are not a part of the original topology
* The current approach is static which requires different changes to be able to stream a different graph, automating the subscription process might be challenging.